commit d4e00dff42892de71594283e5fc9e1d837d65caf
Author: Jim Klimov <jim@jimklimov.com>
Date:   Mon Apr 30 15:13:08 2018 +0200

    Make SMF service name configurable

OI note: PR is merged, this patch-file will be not needed for releases
newer than znapzend-0.18.0

diff --git a/configure.ac b/configure.ac
index 1fc55b4..90fe441 100644
--- a/configure.ac
+++ b/configure.ac
@@ -22,6 +22,8 @@ AC_PATH_PROG(CURL, curl, no)
 AC_PATH_PROG(WGET, wget, no)
 AC_PATH_PROG(POD2MAN, pod2man, no)
 
+# TODO: with some respect to legacy, split such settings to be named
+# after respective service management framework; code supports several
 AC_ARG_ENABLE(svcinstall,
         AC_HELP_STRING(
                 [--enable-svcinstall=DIR],
@@ -33,6 +35,19 @@ AC_ARG_ENABLE(svcinstall,
 AC_SUBST(SVCINSTALLDIR)
 AM_CONDITIONAL(ENABLE_SVCINSTALL,[test $SVCINSTALLDIR != no])
 
+SVCNAME_SMF="oep/znapzend"
+AC_ARG_WITH(svcname-smf,
+        AC_HELP_STRING(
+                [--with-svcname-smf=SMF_FMRI],
+                [use specified SMF FMRI in manifest (defaults to oep/znapzend)]
+        ),
+        [AC_MSG_CHECKING([user-provided SMF service name of '$withval'])
+         SVCNAME_SMF="`echo "$withval" | sed -e 's,^svc:,,' -e 's,^/*,,' -e 's,:.*$,,'`"
+        ],[AC_MSG_CHECKING([default SMF service name])]
+)
+AC_MSG_RESULT([$SVCNAME_SMF])
+AC_SUBST(SVCNAME_SMF)
+
 URL_CAT="neither curl nor wget found"
 if test -x "$WGET"; then
     URL_CAT="$WGET -O -"
diff --git a/init/znapzend.xml.in b/init/znapzend.xml.in
index 2692694..b3ec8f9 100644
--- a/init/znapzend.xml.in
+++ b/init/znapzend.xml.in
@@ -3,7 +3,7 @@
 <service_bundle type="manifest" name="znapzend">
 
 <service 
-        name="oep/znapzend" 
+        name="@SVCNAME_SMF@" 
         type="service" 
         version="4">
 
